### PR TITLE
update apiservice registration namespace and spec on initialization

### PR DESCRIPTION
**Pull request by davidvossel**
related to #932

### CODE REVIEW

**Comment by davidvossel** [comment]
@rmohr I updated this patch to fail early if a namespace mismatch occurs. This puts the resolution on the operator.  The error message clearly states what the issue is and how to resolve it.

File: pkg/virt-api/api.go [hunk 1]
```diff
@@ -494,33 +493,39 @@ func (app *virtAPIApp) createSubresourceApiservice() error {
...
+		_, err = aggregatorClient.ApiregistrationV1beta1().APIServices().Create(newApiService)
 		if err != nil {
 			return err
 		}
-	} else if reflect.DeepEqual(apiService.Spec.CABundle, app.signingCertBytes) == false {
-		// Update apiService if CA bundle doesn't match ours
-		apiService.Spec.CABundle = app.signingCertBytes
+	} else {
+		if apiService.ObjectMeta.Namespace != namespace {
+			return fmt.Errorf("apiservuce [%s] is already registered in a different namespace. Existing apiservice registration must be deleted before virt-api can proceed.", subresourceAggregatedApiName)
```
Comment (rmohr) [inline pkg/virt-api/api.go]: s/apiservuce/apiservice

Comment (davidvossel) [reply pkg/virt-api/api.go]: oops, fixed :)

**Comment by cynepco3hahue** [comment]
retest this please

**Comment by rmohr** [comment]
@cynepco3hahue I think that something goes wrong during the aggregated apiserver registration.

**Comment by cynepco3hahue** [comment]
@rmohr Looks so, I see `virt-api` pod stuck
```
++ kubectl get pods -n kube-system '-ocustom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers
++ awk '!/virt-controller/ && /false/'
++ cluster/kubectl.sh get pods -n kube-system '-ocustom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers
+ '[' -n 'false            virt-api-fd96f94b5-tjskr' ']'
+ echo 'Waiting for KubeVirt containers to become ready ...'
Waiting for KubeVirt containers to become ready ...
+ kubectl get pods -n kube-system '-ocustom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers
+ awk '!/virt-controller/ && /false/'
+ cluster/kubectl.sh get pods -n kube-system '-ocustom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers
false            virt-api-fd96f94b5-tjskr
```

File: pkg/virt-api/api.go [hunk 2]
```diff
@@ -494,33 +493,39 @@ func (app *virtAPIApp) createSubresourceApiservice() error {
...
+	if registerApiService {
+		_, err = aggregatorClient.ApiregistrationV1beta1().APIServices().Create(newApiService)
 		if err != nil {
 			return err
 		}
-	} else if reflect.DeepEqual(apiService.Spec.CABundle, app.signingCertBytes) == false {
-		// Update apiService if CA bundle doesn't match ours
-		apiService.Spec.CABundle = app.signingCertBytes
+	} else {
+		if apiService.ObjectMeta.Namespace != namespace {
```
Comment (rmohr) [inline pkg/virt-api/api.go]: Ha, now I see the issue. That should be:

```golang
if apiService.Spec.Service.Namespace  != namespace {

}

Comment (davidvossel) [reply pkg/virt-api/api.go]: whoa, hooray for functional tests :)

**Pull request merged**